src/checks/cargo/cargo.ts

```diff
@@ -2,6 +2,21 @@ import { touchFiles } from '../../unix'
 import { CheckAnnotation } from '../checks-common'
 import { CargoBuildFinishedMessage, CargoCompilerMessage, CargoMessage, DiagnosticSpan } from './cargo-types'
 
+const DEFAULT_CI_RUSTFLAGS = '-F warnings -D unused'
+
+export function updateEnvRustFlags(ci: boolean): void {
+  let rustFlags = process.env['RUSTFLAGS'] || ''
+
+  if (ci) {
+    rustFlags += ' --cfg ci_build'
+
+    const ciRustFlags = process.env['CI_RUSTFLAGS'] ?? DEFAULT_CI_RUSTFLAGS
+    rustFlags += ` ${ciRustFlags}`
+  }
+
+  process.env['RUSTFLAGS'] = rustFlags.trim()
```

Comment:
I would probably prefer to add environment variable when running the cargo process instead of setting it globally for the check code, but I will merge, if you feel for it you can do another pull to fix this.

Reply:
Just to clarify, in this case by _adding_ it, you mean passing it as `options` to `run(Json)Command`, right? :)

(Created a story [ch-55542](https://app.clubhouse.io/connectedcars/story/55542/pass-env-to-child-process-instead-of-overriding-global-env-vars))

Reply:
Yes, so it's only the sub process that gets its environment set. For to be security conscious it always a good idea to set the environment so you don't expose random environments that might otherwise end up in logs where they should not go, fx. in this case we have some tokens and passwords that the cargo command does not need.

Reply:
Of course, just wanted to be sure :)